src/a5dd0_len_114e0.c

```diff
@@ -2988,8 +2988,638 @@ void state_render_frontUI(void) {
+        combineSubType = 1;
+    }
+
+    // fog mode
+    switch ((u32)(model->customGfxIndex >> 4)) {
+        case FOG_MODE_1:
+            s3 += 3;
+            combineSubType = 2;
+            break;
+        case FOG_MODE_2:
+            if (renderMode <= RENDER_MODE_ALPHATEST_NO_ZB) {
+                gDPSetPrimColor((*gfxPos)++, 0, 0, mdl_renderModelFogPrimColorR,
+                                                   mdl_renderModelFogPrimColorG,
+                                                   mdl_renderModelFogPrimColorB,
+                                                   mdl_renderModelFogPrimColorA);
+                gDPSetFogColor((*gfxPos)++, mdl_renderModelFogColorR,
+                                            mdl_renderModelFogColorG,
+                                            mdl_renderModelFogColorB, 0);
+                gSPFogPosition((*gfxPos)++, mdl_renderModelFogStart, mdl_renderModelFogEnd);
+                s3 += 9;
+                combineSubType = 3;
+            }
+            break;
+        case FOG_MODE_3:
+            s3 = 2;
+            combineSubType = 4;
+            gDPSetPrimColor((*gfxPos)++, 0, 0, gRenderModelPrimR,
+                                               gRenderModelPrimG,
+                                               gRenderModelPrimB, 255);
+            gDPSetEnvColor((*gfxPos)++, gRenderModelEnvR,
+                                        gRenderModelEnvG,
+                                        gRenderModelEnvB, 255);
+            break;
+    }
+
+    gDPPipeSync((*gfxPos)++);
+
+    if (model->groupData != NULL) {
+        Lightsn* lightningGroup = model->groupData->lightingGroup;
+        if (model->groupData->lightingGroup != NULL) {
+            switch (model->groupData->numLights) {
+                case 0:
+                    gSPSetLights0((*gfxPos)++, lightningGroup[0]);
+                    break;
+                case 1:
+                    gSPSetLights1((*gfxPos)++, lightningGroup[0]);
+                    break;
+                case 2:
+                    gSPSetLights2((*gfxPos)++, lightningGroup[0]);
+                    break;
+                case 3:
+                    gSPSetLights3((*gfxPos)++, lightningGroup[0]);
+                    break;
+                case 4:
+                    gSPSetLights4((*gfxPos)++, lightningGroup[0]);
+                    break;
+                case 5:
+                    gSPSetLights5((*gfxPos)++, lightningGroup[0]);
+                    break;
+                case 6:
+                    gSPSetLights6((*gfxPos)++, lightningGroup[0]);
+                    break;
+                case 7:
+                    gSPSetLights7((*gfxPos)++, lightningGroup[0]);
+                    break;
+            }
+        }
+    }
+
+    if (textureHeader != NULL) {
+        switch (extraTileType) {
+            case EXTRA_TILE_3:
+            case EXTRA_TILE_4:
+                prop = get_model_property(modelNode, MODEL_PROP_KEY_SPECIAL);
```

Comment:
The two-s32 values for this property are:
00AAABBB 000CD0EF
A = auxOffsetT
B = auxOffsetS
C = auxShiftT
D = auxShiftS
E = ModelReplaceType (an enum)
F = defaultScrollUnit